include/core/evaporation_model.h

```diff
@@ -0,0 +1,683 @@
+
+    return mass_flux_value;
+  }
+
+  /**
+   * @brief vector_mass_flux Calculates the values of the evaporation mass flux
+   * for multiple points
+   * @param field_vectors Value of the various fields on which the flux may depend.
+   * @param mass_flux_vector Vectors of the mass flux values
+   */
+  void
+  vector_mass_flux(const std::map<field, std::vector<double>> &field_vectors,
+                   std::vector<double> &mass_flux_vector) override
+  {
+    const std::vector<double> &temperature =
+      field_vectors.at(field::temperature);
+
+    const unsigned int n_pts = mass_flux_vector.size();
+
+    std::vector<double> saturation_pressure_vector(n_pts);
+    vector_saturation_pressure(field_vectors, saturation_pressure_vector);
+
+    for (unsigned int i = 0; i < mass_flux_vector.size(); ++i)
+      {
+        const double temperature_inv = 1.0 / temperature[i];
+
+        mass_flux_vector[i] = evaporation_coefficient *
+                              saturation_pressure_vector[i] *
+                              std::sqrt(M_x_2PI_inv_x_R_inv * temperature_inv);
+      }
+  }
+
+  /**
+   * @brief heat_flux Calculates the value of the evaporation heat flux.
+   * @param fields_value Value of the various field on which the flux may depend.
+   * @return value of the heat flux calculated with the fields_value
+   */
+  double
+  heat_flux(const std::map<field, double> &fields_value) override
+  {
+    const double mass_flux_value = mass_flux(fields_value);
+
+    return mass_flux_value * latent_heat_evaporation;
+  }
+
+  /**
+   * @brief vector_heat_flux Calculates the values of the evaporation heat flux
+   * for multiple points
+   * @param field_vectors Value of the various fields on which the flux may depend.
+   * @param heat_flux_vector Vectors of the heat flux values
+   */
+  void
+  vector_heat_flux(const std::map<field, std::vector<double>> &field_vectors,
+                   std::vector<double> &heat_flux_vector) override
+  {
+    const unsigned int n_pts = heat_flux_vector.size();
+
+    std::vector<double> mass_flux_vector(n_pts);
+    vector_mass_flux(field_vectors, mass_flux_vector);
+
+    for (unsigned int i = 0; i < n_pts; ++i)
+      {
+        heat_flux_vector[i] = mass_flux_vector[i] * latent_heat_evaporation;
+      }
+  }
+
+  /**
+   * @brief heat_flux_jacobian Calculates the jacobian (the partial derivative)
+   * of the evaporation heat flux with respect to a field
+   * @param field_values Value of the various fields on which the flux may depend.
+   * @param id Indicator of the field with respect to which the jacobian
+   * should be calculated
+   * @return value of the partial derivative of the heat flux with respect to the field.
+   */
+  double
+  heat_flux_jacobian(const std::map<field, double> &fields_value,
+                     const field                    id) override
+  {
+    const double temperature_inv = 1.0 / fields_value.at(field::temperature);
```

Comment:
I would double check on the temperature to make sure the inverse is at least well-defined no? So many things can go wrong if the initial condition is not set-up correctly